deepchem/models/losses.py

```diff
@@ -465,6 +470,216 @@ def loss(inputs):
         return loss
 
 
+class MolGraphConvMILoss():
+    """
+    Losses used to maximize the mutual information between molecular graph encodings.
+    Encodings can be generated from molecular graphs using a graph convolutional network, and the mutual information is calculated between subgraphs (local encodings) and the entire molecule (global encodings) in order to enable unsupervised training.
+
+    This class supports only PyTorch.
+
+    Parameters:
+    device: torch.device
+        Device to run the model on.
+    measure: str
+        The divergence measure to use for the unsupervised loss. Options are 'GAN', 'JSD',
+        'KL', 'RKL', 'X2', 'DV', 'H2', or 'W1'.
+    average_loss: bool
+        Whether to average the loss over the batch
+    """
+
+    def __init__(self, device, measure='JSD', average_loss=True):
+        self.device = device
+        self.measure = measure
+        self.average_loss = average_loss
+
+    def local_global_loss(self, l_enc, g_enc, index):
+        """
+        Compares the local and global encodings of molecular graphs and returns the loss between them based on the measure specified.
+        Global in this case refers to the global encoding of the entire molecule, and local refers to the encoding of subgraphs.
+
+        Parameters:
+        ----------
+        l_enc: torch.Tensor
+            Local feature map from the encoder.
+        g_enc: torch.Tensor
+            Global features from the encoder.
+        index: torch.Tensor
+            Index of the graph that each node belongs to.
+        device: torch.device
```

Comment:
Some of these are no longer arguments for this function and can be removed (device/measure/etc)